extensions/sql-database-projects/src/typings/vscode-mssql.d.ts

```diff
@@ -92,12 +92,12 @@ declare module 'vscode-mssql' {
         /**
          * email
          */
-        email: string;
+        email?: string;
```

Comment:
Please update https://github.com/microsoft/vscode-mssql/blob/main/typings/vscode-mssql.d.ts first before making changes here. Although you should also be checking that not having these isn't going to break it - this interface was being used internally there and so they may be expecting these all to exist.

Reply:
all these fields should be optional. there are problem with passing empty string to some field that make connection fail. I'll do another fix later with vscode side change

Reply:
You still should be updating the VS Code MSSQL extension typings file first. This is a direct copy of that so having them not be in sync is problematic.

Reply:
sure, I tried optional in vscode but didn't work so I made them accept undefiled. in vscode I see the field are assiged to undefined in some places but the type is not string | undefined. So I fixed it there